File: week2_algorithic_warm-up/fibonacci_sum_last_digit.cpp
#include <iostream>


int fibonacciSumLastDigit(int);

int main()
{
    int n;
    std::cin >> n;
    std::cout << fibonacciSumLastDigit(n) << '\n'; 
}

int fibonacciSumLastDigit(int n)
{
    //This function compute the last digit of the sum of the fibonacci number
    if (n <= 0)
        return 0;

    int fib[n+1];
    fib[0] = 0 , fib[1] = 1;

    int sum = fib[0] + fib[1];

    for (int i=2; i <= n; i++)
    {
        fib[i] = fib[i-1] + fib[i-2];
        sum += fib[i];
    }

    return sum % 10;
}

File: week2_algorithic_warm-up/fibonacci_sum_last_digit/fibonacci_sum_last_digit_python/fibonacci_sum_last_digit.py
from sys import stdin

def fibonacci_sum_last_digit(n):
    if n <= 0:
        return 0

    fib = [int(i) for i in range(n+1)]

    fib[0] = 0
    fib[1] = 1

    sum = fib[0] + fib[1]

    for i in range(2, n+1):
        fib[i] = fib[i-1] + fib[i-2]
        sum += fib[i]

    return sum % 10


n = int(input())
print(fibonacci_sum_last_digit(n))
